## Non Graded Challenge 6

In [1]:
#calling necessary library
import numpy as np

Apakah vektor (1, 3) adalah eigenvector dari [[1, -1], [6, 4]], jika iya berapakah eigenvalue-nya?

dapat dikatakan eigen value jika:
A - (eigenvalue * I) = 0, maka

In [3]:
#Matrice A
A = np.array([[1,-1], [6,4]])

#Vector B
b = np.array([1, 3])

#Finding eigen value and vector
u, v = np.linalg.eig(A)

print('eigen value:\n',u)
print('\neigen vector:\n',v)

eigen value:
 [2.5+1.93649167j 2.5-1.93649167j]

eigen vector:
 [[-0.23145502+0.29880715j -0.23145502-0.29880715j]
 [ 0.9258201 +0.j          0.9258201 -0.j        ]]


Matrice v above composed by 2 eigen vectors, hence we could check the equation: A*(eigen vector) = eigen value * eigen vector

we will use the last element from eigen value array and eigen vector array

In [13]:
#w and x eigen vectors from column 1 and 2
w = v[:,1]
x = v[:,0]

i = np.dot(A, b)
j = np.dot(A,w)

print('A dot b:\n',i)
print('\nmatrice A * eigen vector 1:\n',j)
print('\nmatrice A * eigen vector 2:\n',np.dot(A,w))



A dot b:
 [-2 18]

matrice A * eigen vector 1:
 [-1.15727512-0.29880715j  2.31455025-1.79284291j]

matrice A * eigen vector 2:
 [-1.15727512-0.29880715j  2.31455025-1.79284291j]


karena hasil i dan j tidak sama, maka vector b bukan eigen vector dari matriks A

dan eigen value yang didapatkan adalah : 2.5+1.93649167j dan 2.5-1.93649167j